pkg/apis/monitoring/v1/types.go

```diff
@@ -104,6 +105,10 @@ type ProxyConfig struct {
 	ProxyConnectHeader map[string]v1.SecretKeySelector `json:"proxyConnectHeader,omitempty"`
 }
 
+func IsProxyConfigNotEqualToZero(pc ProxyConfig) bool {
```

Comment:
I'm not a fan of including utility functions in the pkg/apis/monitoring module. I think that it could live in pkg/prometheus. But we could as well use reflect.ValueOf(proxyConfig).IsZero() directly (which would be more expressive IMHO).